### PR TITLE
[tune] Allow resources to not all be assigned to the driver

### PR DESCRIPTION
**Pull request by ericl**
Currently all requested resources get assigned to the driver actor. However, sometimes those resources should actually be assigned to actors created by the driver.

@richardliaw @pcmoritz

### CODE REVIEW

**Comment by ericl** [comment]
ping

File: python/ray/tune/trial.py [hunk 1]
```diff
@@ -10,8 +10,31 @@
 from ray.rllib.agent import get_agent_class
 
 
-# Ray resources required to schedule a Trial
-Resources = namedtuple("Resources", ["cpu", "gpu"])
+class Resources(
+        namedtuple("Resources", ["cpu", "gpu", "driver_cpu", "driver_gpu"])):
```
Comment (richardliaw) [inline python/ray/tune/trial.py]: why do we want to keep this as a named tuple?

**Review by richardliaw — CHANGES_REQUESTED** [review]
`trial_runner.has_resources` needs to also be updated so that this is included in the bookkeeping. Looks good otherwise.

File: python/ray/tune/trial.py (hunk 1 above)
Comment (ericl) [reply python/ray/tune/trial.py]: Hm that gives you some nice properties for free, basically it's like a struct as a base class.

**Comment by ericl** [comment]
Renamed to *_limit as discussed offline.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
@ericl, I made a few minor changes. Will merge after tests pass (excluding #1165).

**Pull request merged**